applications/incompressible_navier_stokes/navier_stokes_manufactured/application.h

```diff
@@ -0,0 +1,798 @@
+    		             generalized_newtonian_lambda,
+      					 "Generalized Newtonian models: lambda.",
+      					 dealii::Patterns::Double());
+      prm.add_parameter("GeneralizedNewtonianA",
+    		             generalized_newtonian_a,
+      					 "Generalized Newtonian models: a.",
+      					 dealii::Patterns::Double());
+      prm.add_parameter("GeneralizedNewtonianN",
+    		             generalized_newtonian_n,
+      					 "Generalized Newtonian models: n.",
+      					 dealii::Patterns::Double());
+    prm.leave_subsection();
+    // clang-format on
+  }
+
+private:
+  void
+  parse_parameters() final
+  {
+    ApplicationBase<dim, Number>::parse_parameters();
+
+    // clang-format off
+    if     (temporal_discretization_string == "BDFCoupledSolution")     temporal_discretization = TemporalDiscretization::BDFCoupledSolution;
+    else if(temporal_discretization_string == "BDFPressureCorrection")  temporal_discretization = TemporalDiscretization::BDFPressureCorrection;
+    else if(temporal_discretization_string == "BDFDualSplittingScheme") temporal_discretization = TemporalDiscretization::BDFDualSplittingScheme;
+    else AssertThrow(false, dealii::ExcMessage("Unknown temporal discretization. Not implemented."));
+    // clang-format on
+  }
+  void
+  set_parameters() final
+  {
+    // MATHEMATICAL MODEL
+    this->param.problem_type = ProblemType::Unsteady;
+    if(include_convective_term == true)
+      this->param.equation_type = EquationType::NavierStokes;
+    else
+      this->param.equation_type = EquationType::Stokes;
+
+    this->param.formulation_convective_term = FormulationConvectiveTerm::ConvectiveFormulation;
+
+    FormulationViscousTerm formulation_viscous_term =
+      formulation_viscous_term_laplace_formulation ? FormulationViscousTerm::LaplaceFormulation :
+                                                     FormulationViscousTerm::DivergenceFormulation;
+    this->param.formulation_viscous_term = formulation_viscous_term;
+    this->param.right_hand_side          = true;
+
+
+    // PHYSICAL QUANTITIES
+    this->param.start_time = start_time;
+    this->param.end_time   = end_time;
+    this->param.viscosity  = kinematic_viscosity;
+    this->param.density    = density;
+
+
+    // TEMPORAL DISCRETIZATION
+    this->param.solver_type                   = SolverType::Unsteady;
+    this->param.temporal_discretization       = temporal_discretization;
+    this->param.calculation_of_time_step_size = TimeStepCalculation::UserSpecified;
+    this->param.time_step_size                = this->param.end_time;
+    this->param.order_time_integrator         = 2;     // 1; // 2; // 3;
+    this->param.start_with_low_order          = false; // true;
+
+    // output of solver information
+    this->param.solver_info_data.interval_time =
+      (this->param.end_time - this->param.start_time) / 10;
+
+
+    // SPATIAL DISCRETIZATION
+    this->param.grid.triangulation_type = TriangulationType::Distributed;
```

Comment:
I think fully-distributed will not work for the present `create_grid()` function, so maybe add this here as a comment.

Reply:
I changed the `create_grid()` accordingly (and tested it)!